rpc/message.go

```diff
@@ -89,6 +91,47 @@ func (r rpcCallMessage) Err() error {
 	return r.err
 }
 
+func (r rpcCallMessage) Compression() CompressionType {
+	return CompressionNone
+}
+
+type rpcCallCompressedMessage struct {
+	rpcCallMessage
+	ctype CompressionType
+}
+
+func (rpcCallCompressedMessage) MinLength() int {
+	return 4
+}
+
+func (r *rpcCallCompressedMessage) DecodeMessage(l int, d *fieldDecoder, p *protocolHandler, _ *callContainer, _ *compressorCacher) error {
+	if r.err = d.Decode(&r.seqno); r.err != nil {
+		return r.err
+	}
+	if r.err = d.Decode(&r.ctype); r.err != nil {
```

Comment:
Does this work? The struct order matters when decoding this stuff: `rpcCallMessage` has `seqno`, `name`, `arg`, in that order, and must be decoded in that order. Since you embedded an `rpcCallMessage` into the `rpcCallCompressedMessage`, then you must decode them in the same order. I also think `r.err` will mess you up here.

You might need to specify the argument explicitly instead of embedding `rpcCallMessage` in `rpcCallCompressedMessage`. I'd do it like this:

```go
type rpcCallCompressedMessage struct {
	basicRPCData
	seqno SeqNumber
	ctype CompressionType
	name  string
	arg   interface{}
	err   error
}
```
And then you shouldn't have to change this code.

Reply:
Right, ok